File: C++/B/57.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n, m, e, ans = 1;
    cin >> n >> m;
    vector<int> a(n), b(n);
    vector<int> c(m), d(m);
    for (int i = 0; i < n; i++)
    {
        cin >> a[i] >> b[i];
    }
    for (int i = 0; i < m; i++)
    {
        cin >> c[i] >> d[i];
    }

    for (int i = 0; i < n; i++)
    {
        e = abs(a[i] - c[0]) + abs(b[i] - d[0]);
        for (int j = 1; j < m; j++)
        {
            if (e > abs(a[i] - c[j]) + abs(b[i] - d[j]))
            {
                e = abs(a[i] - c[j]) + abs(b[i] - d[j]);
                ans = j + 1;
            }
        }
        cout << ans << endl;
        ans = 1;
    }
}

File: ruby/B/57.rb
n, m = gets.split.map(&:to_i)
arr1, arr2, arr3, arr4 = [], [], [], []
ans = 1
n.times do
  a, b = gets.split.map(&:to_i)
  arr1 << a
  arr2 << b
end
m.times do
  c, d = gets.split.map(&:to_i)
  arr3 << c
  arr4 << d
end
for i in 0..n - 1
  md = (arr1[i] - arr3[0]).abs + (arr2[i] - arr4[0]).abs
  for j in 1..m - 1
    if (arr1[i] - arr3[j]).abs + (arr2[i] - arr4[j]).abs < md
      md = (arr1[i] - arr3[j]).abs + (arr2[i] - arr4[j]).abs
      ans = j + 1
    end
  end
  p ans
  ans = 1
end
